src/librustc_trans/trans/base.rs

```diff
@@ -1685,6 +1685,7 @@ impl<'blk, 'tcx> FunctionContext<'blk, 'tcx> {
                     for (j, &tupled_arg_ty) in tupled_arg_tys.iter().enumerate() {
                         let dst = StructGEP(bcx, llval, j);
                         let arg = &self.fn_ty.args[idx];
+                        idx += 1;
```

Comment:
Huh, did I forget this in both places? I know @nikomatsakis noticed this was missing on the MIR side.